File: Filtering in MVC 3/[VB]-Filtering in MVC 3/VB/Models/EF_UserRepository.vb
Namespace Mvc3Filter.Models
	Public Class EF_UserRepository
		Implements Mvc3Filter.Models.IUserDB

		Private _db As New UserDatabaseEntities()

		#Region "IUserDB Members"

		Public Function UserExists(ByVal UserName As String) As Boolean Implements Mvc3Filter.Models.IUserDB.UserExists
			Dim usr = _db.Users.FirstOrDefault(Function(d) d.UserName = UserName)
			Return (usr IsNot Nothing)
		End Function

        Public Property GetAllUsers() As List(Of User) Implements Mvc3Filter.Models.IUserDB.GetAllUsers
            Get
                Return _db.Users.ToList()
            End Get
            Set(ByVal value As List(Of User))

            End Set
        End Property

		Public Sub CreateNewUser(ByVal newUser As User) Implements Mvc3Filter.Models.IUserDB.CreateNewUser
			_db.AddToUsers(newUser)
			_db.SaveChanges()
		End Sub

		Public Function GetUser(ByVal uid As String) As User Implements Mvc3Filter.Models.IUserDB.GetUser
			Return _db.Users.FirstOrDefault(Function(d) d.UserName = uid)
		End Function

		Public Sub Remove(ByVal usrName As String) Implements Mvc3Filter.Models.IUserDB.Remove
			Dim userToRemove = GetUser(usrName)
			_db.Users.DeleteObject(userToRemove)
			_db.SaveChanges()
		End Sub

		Public Sub Update(ByVal userToUpdate As User) Implements Mvc3Filter.Models.IUserDB.Update
			_db.SaveChanges()
		End Sub

		#End Region
	End Class
End Namespace

File: Filtering in MVC 3/[C#]-Filtering in MVC 3/C#/Mvc3Filter/Models/EF_UserRepository.cs
using System.Collections.Generic;
using System.Linq;

namespace Mvc3Filter.Models {
    public class EF_UserRepository : Mvc3Filter.Models.IUserDB {

        private UserDatabaseEntities _db = new UserDatabaseEntities();

        #region IUserDB Members

        public bool UserExists(string UserName) {
            var usr = _db.Users.FirstOrDefault(d => d.UserName == UserName);
            return (usr != null);
        }

        public List<User> GetAllUsers {
            get { return _db.Users.ToList(); }
        }

        public void CreateNewUser(User newUser) {
            _db.AddToUsers(newUser);
            _db.SaveChanges();
        }

        public User GetUser(string uid) {
            return _db.Users.FirstOrDefault(d => d.UserName == uid);
        }

        public void Remove(string usrName) {
            var userToRemove = GetUser(usrName);
            _db.Users.DeleteObject(userToRemove);
            _db.SaveChanges();
        }

        public void Update(User userToUpdate) {
            _db.SaveChanges();
        }

        #endregion
    }
}